### PR TITLE
Add M1 and M2 button indicators to HUD

### PR DESCRIPTION
**Pull request by Hat-Kid**
This adds button indicators for the left and right mouse buttons to the HUD like this:
unknown

### CODE REVIEW

File: mp/src/game/client/momentum/ui/HUD/hud_keypress.cpp [hunk 1]
```diff
@@ -73,8 +73,12 @@ class CHudKeyPressDisplay : public CHudElement, public Panel
     wchar_t m_pwJump[BUFSIZELOCL];
     wchar_t m_pwDuck[BUFSIZELOCL];
     wchar_t m_pwStrafe[BUFSIZELOCL];
+    wchar_t m_pwM1[BUFSIZELOCL];
+    wchar_t m_pwM2[BUFSIZELOCL];
     float m_fJumpColorUntil;
     float m_fDuckColorUntil;
+    float m_fM1ColorUntil;
+    float m_fM2ColorUntil;
```
Comment (Gocnak) [inline mp/src/game/client/momentum/ui/HUD/hud_keypress.cpp]: Why are these added? You don't seem to use them.

File: mp/src/game/client/momentum/ui/HUD/hud_keypress.cpp [hunk 2]
```diff
@@ -221,6 +228,21 @@ void CHudKeyPressDisplay::Paint()
         surface()->DrawSetTextPos(GetTextCenter(m_hWordTextFont, m_pwDuck), duck_row_ypos);
         surface()->DrawPrintText(m_pwDuck, wcslen(m_pwDuck));
     }
+    // Add M1 and M2 buttons
+    if (nButtons & IN_ATTACK)
+    {
+        CHECK_INPUT_P(IN_ATTACK);
+        int text_m1 = GetTextCenter(m_hTextFont, m_pwM1) - (UTIL_ComputeStringWidth(m_hTextFont, m_pwM1)) * 3.5;
```
Comment (Gocnak) [inline mp/src/game/client/momentum/ui/HUD/hud_keypress.cpp]: The variable name here is not descriptive, use something like `xposM1` instead.

Also I'm not a fan of the random `* 3.5` scalar happening here. Is there any way this can be more absolute? It's only two characters, this seems like it's overly complicated for positioning them.

File: mp/src/game/client/momentum/ui/HUD/hud_keypress.cpp [hunk 3]
```diff
@@ -221,6 +228,21 @@ void CHudKeyPressDisplay::Paint()
...
+        surface()->DrawSetTextPos(text_m1, top_row_ypos);
+        surface()->DrawPrintText(m_pwM1, wcslen(m_pwM1));
+    }
+    if (nButtons & IN_ATTACK2)
+    {
+        CHECK_INPUT_P(IN_ATTACK2);
+        int text_m2 = GetTextCenter(m_hTextFont, m_pwM2) + (UTIL_ComputeStringWidth(m_hTextFont, m_pwM2)) * 2.75;
```
Comment (Gocnak) [inline mp/src/game/client/momentum/ui/HUD/hud_keypress.cpp]: Same two comments as above for m1

File: mp/src/game/client/momentum/ui/HUD/hud_keypress.cpp [hunk 4]
```diff
@@ -324,11 +346,19 @@ void CHudKeyPressDisplay::DrawKeyTemplates()
...
     // duck
     CHECK_INPUT_N(IN_DUCK);
     surface()->DrawSetTextPos(GetTextCenter(m_hWordTextFont, m_pwDuck), duck_row_ypos);
     surface()->DrawPrintText(m_pwDuck, wcslen(m_pwDuck));
+    CHECK_INPUT_N(IN_ATTACK);
+    int text_m1 = GetTextCenter(m_hTextFont, m_pwM1) - (UTIL_ComputeStringWidth(m_hTextFont, m_pwM1)) * 3.5;
+    surface()->DrawSetTextPos(text_m1, top_row_ypos);
+    surface()->DrawPrintText(m_pwM1, wcslen(m_pwM1));
+    CHECK_INPUT_N(IN_ATTACK2);
+    int text_m2 = GetTextCenter(m_hTextFont, m_pwM2) + (UTIL_ComputeStringWidth(m_hTextFont, m_pwM2)) * 2.75;
```
Comment (Gocnak) [inline mp/src/game/client/momentum/ui/HUD/hud_keypress.cpp]: I'm starting to wonder if there's a refactor opportunity here to store keys (including their templates) in their own class and do all this calculation inside of them...

The reason I bring this up is I'm seeing very similar positioning code being calculated here as in Paint, and not just for your added keys.

File: mp/src/game/client/momentum/ui/HUD/hud_keypress.cpp (hunk 2 above)
Comment (Hat-Kid) [reply mp/src/game/client/momentum/ui/HUD/hud_keypress.cpp]: The issue with the positioning was that I didn't really have a sense of scale for how much changing the value would change the x position of the element so I just experimented with it until I reached something that looked good enough. I will try to figure out a way to define it better.

Comment (Hat-Kid) [reply mp/src/game/client/momentum/ui/HUD/hud_keypress.cpp]: I found out that using `WordTextFont` instead of `TextFont` makes the positioning more consistent without the need for arbitrary factors (using one single factor across all elements is sufficient).
